### PR TITLE
Add more `LinkedHashMap` functionality

### PR DESCRIPTION
**Pull request by apasel422**
CC #127.

### CODE REVIEW

File: src/linked_hash_map.rs [hunk 1]
```diff
@@ -71,6 +72,22 @@ impl<K: PartialEq> PartialEq for KeyRef<K> {
 
 impl<K: Eq> Eq for KeyRef<K> {}
 
+// This type exists only to support borrowing `KeyRef`s, which cannot be borrowed to `Q` directly
+// due to conflicting implementations of `Borrow`. The layout of `&Qey<Q>` must be identical to
+// `&Q` in order to support transmuting in the `Qey::from_ref` method.
```
Comment (Gankro) [inline src/linked_hash_map.rs]: This is getting into some hairy shit...

Comment (reem) [inline src/linked_hash_map.rs]: A little hacky.

Comment (apasel422) [inline src/linked_hash_map.rs]: Agreed, but this entire module is pretty hacky. Presumably a safer version of this collection could be written if it had access to the internal representation of `HashMap`.

**Comment by apasel422** [comment]
Re-targeting against external crate.

**Comment by Gankro** [comment]
Oh whoops, forgot about this PR, sorry!

**Pull request closed without merging**